pkg/controller/batchrelease/batchrelease_controller.go

```diff
@@ -184,12 +182,9 @@ func (r *BatchReleaseReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return reconcile.Result{}, err
 	}
 
-	// set the release info for executor before executing.
-	r.executor.SetReleaseInfo(release)
-
 	// executor start to execute the batch release plan.
 	startTimestamp := time.Now()
-	result, currentStatus, err := r.executor.Do()
+	result, currentStatus, err := NewReleasePlanExecutor(r.Client, r.recorder, release).Do()
```

Comment:
is it possible to add release as a variable to Do and deepcopy release status if needed,  so we can still share the same executor?